src/app/components/app.tsx

```diff
@@ -50,56 +51,92 @@ interface IAppProps {
 
 export const App = (props: IAppProps) => {
   const {interactiveState, readOnly} = props;
+  const [readAloudMode, setReadAloudMode] = useState<boolean>(false);
+  const [isAudioPlaying, setIsAudioPlaying] = useState<boolean>(false);
+  const [activeAudioId, setActiveAudioId] = useState<string>("");
+
+  const { startSimulation, endSimulation, isRunning } = useSimulationRunner();
+
+  const t = useCallback((string: string) => {
```

Comment:
This might be better encapsulated as a hook that returns the `t()` function and keeps all the state internal (readAloudMode, isAudioPlaying, activeAudioId).  Then you wouldn't need to pass it as a property to the child elements.

Reply:
To do that you could want to create a top level React context to hold the state and then a hook that returns the `t()` function in the context.

Reply:
I was wondering about that as an approach. I will try to re-factor it this way. I have a few more changes to make too and will be re-requesting your review later today. Thanks Doug :)

Reply:
Hi Doug, I re-factored the translate function into the useTranslation hook. However, I still ended up passing it down to the components since the hook has a dependency on the "isRunning" variable. Please let me know if you think there might be a better way to implement it.